app/hallebarde/infrastructure/exchange_repository.py

```diff
@@ -1,22 +1,41 @@
-import boto3
+from typing import List, Optional
 
+import boto3
+from boto3.dynamodb.table import TableResource
+from boto3.dynamodb.conditions import Attr
 import hallebarde.config
 from hallebarde.domain.exchange import Exchange
 
 
 def save(exchange: Exchange) -> None:
     table = _get_dynamodb_table()
-    table.put_item(Item={'identifier': exchange.identifier, 'upload_token': exchange.upload_token,
-                         'download_token': exchange.download_token})
+    table.put_item(
+        Item={'identifier': exchange.identifier, 'email': exchange.email, 'upload_token': exchange.upload_token,
```

Comment:
Would it work with `Item=exchange.__dict__` ?

Reply:
I think so, let me test